pkg/proc/proc.go

```diff
@@ -588,6 +567,53 @@ func GoroutinesInfo(dbp Process, start, count int) ([]*G, int, error) {
 	return allg, -1, nil
 }
 
+func getAllGs(dbp Process) (uint64, uint64, error) {
```

Comment:
This function technically only needs a `*BinaryInfo` struct and a `MemoryReadWriter`. Instead of passing in the whole `Process` let's just pass in `dbp.BinInfo()` and `dbp.CurrentThread()`.

Reply:
Also, shouldn't this be the level in which we maintain the allgcache? This gets called from both `GoroutinesInfo` and `FindGoroutine`. We could cache the address of runtime.allg as well. This would offer significant improvement as we won't need to issue a bunch of calls to read process memory, dig through debug info, etc.